File: entities/registration.go
package entities

import (
	"time"

	"labix.org/v2/mgo/bson"
)

type Registration struct {
	Id           bson.ObjectId `bson:"_id"`
	EventName    string        `bson:"event_name,omitempty"`
	CallbackUrl  string        `bson:"callback_url,omitempty"`
	CreationDate time.Time     `bson:"creation_date,omitempty"`
}

func CreateNewRegistration(event string, callback string) Registration {
	return Registration{
		Id:           bson.NewObjectId(),
		EventName:    event,
		CallbackUrl:  callback,
		CreationDate: time.Now(),
	}
}

File: _minke/features/models/registration.rb
class Registration
  include Mongoid::Document
  field :_id,           type: Moped::BSON::ObjectId
  field :event_name,  type: String
  field :callback_url,  type: String
  field :creation_date, type: DateTime
end

FactoryGirl.define do
  factory :registration, class: Registration do
    event_name "users.new_email"
    callback_url "http://myserver.com/v1/newemail"
    creation_date Time.now
  end
end
